gopls: Format on save is disabled globally for the duration of the session after syntax error

# System information
macOS Catalina 10.15.5 (19F101)
Sublime Text Dev Channel, Build 4074

# Install
Package Control

# Reproduction
Assuming the LSP configuration
```json
{
    "lsp_format_on_save": true,
    "lsp_code_actions_on_save": {
        "source.fixAll": true,
        "source.organizeImports": true
    },
    "show_symbol_action_links": true,
    "only_show_lsp_completions": true,
    "clients": {
        "gopls": {
            "enabled": true,
            "settings": {
                "gopls.usePlaceholders": true,
                "gopls.analyses": {
                  "fillreturns": true,
                  "nonewvars": true,
                  "undeclaredname": true,
                  "unusedparams": true
                },
                "gopls.completeUnimported": true,
                "gopls.staticcheck": true
            }
        }
    }
}
```

Then, given the snippet (note the superfluous indentation, and missing `.`)
```go
package main

import "fmt"

func main() {



                fmt Println("Hello, world!")
}
```

Save the snippet as `main.go`, open `main.go` in `subl` and save (to run the format on save).

Then replace
```go
fmt Println("Hello, world!")
```
with
```go
fmt.Println("Hello, world!")
```

From this point, saving the file does not do any formatting. However, it does still do imports, if
```go
import "fmt"
```
is removed.

# Logs
```
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 795, in on_activated_async
    run_view_callbacks('on_activated_async', view_id)
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 622, in run_view_callbacks
    callback(*args)
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 167, in exception_handler
    return event_handler(*args)
  File "/Users/thomas/Library/Application Support/Sublime Text 3/Installed Packages/LSP.sublime-package/plugin/core/documents.py", line 52, in on_activated_async
    self.manager.activate_view(self.view)
  File "/Users/thomas/Library/Application Support/Sublime Text 3/Installed Packages/LSP.sublime-package/plugin/core/windows.py", line 419, in activate_view
    self._initialize_on_open(view)
  File "/Users/thomas/Library/Application Support/Sublime Text 3/Installed Packages/LSP.sublime-package/plugin/core/windows.py", line 461, in _initialize_on_open
    startable_configs = needed_configs(self._configs.match_view(view))  # type: ignore
  File "/Users/thomas/Library/Application Support/Sublime Text 3/Installed Packages/LSP.sublime-package/plugin/core/windows.py", line 441, in needed_configs
    for c in configs:
  File "/Users/thomas/Library/Application Support/Sublime Text 3/Installed Packages/LSP.sublime-package/plugin/core/configurations.py", line 47, in match_view
    configs = self.match_document(view2scope(view))
  File "/Users/thomas/Library/Application Support/Sublime Text 3/Installed Packages/LSP.sublime-package/plugin/core/types.py", line 115, in view2scope
    return view.scope_name(0).split()[0]
IndexError: list index out of range
```

For the record, gopls sends this response when there is a syntax error in the document.
```json
{
  "jsonrpc": "2.0",
  "error": {
    "data": null,
    "code": 0,
    "message": "computing fix edits: /Users/me/main.go:9:21: expected ';', found Println (and 2 more errors)"
  },
  "id": 3,
  "result": null
}
```
Which is invalid per spec as, as @rwols said, the `result` property is not allowed on error responses.

--- a/plugin/core/rpc.py
+++ b/plugin/core/rpc.py
@@ -106,6 +106,7 @@ class SyncRequestStatus:
 
     def reset(self) -> None:
         self.__payload = None
+        self.__error = None
         self.__request_id = -1
         self.__response_id = -1
 

